src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/RSAPKCS1SignatureDeformatter.cs

```diff
@@ -33,10 +33,16 @@ public override void SetHashAlgorithm(string strName)
             try
             {
                 // Verify the name
-                Oid.FromFriendlyName(strName, OidGroup.HashAlgorithm);
-
-                // Uppercase known names as required for BCrypt
-                _algName = HashAlgorithmNames.ToUpper(strName);
+                if (CryptoConfig.MapNameToOID(strName) != null)
+                {
+                    // Uppercase known names as required for BCrypt
+                    _algName = HashAlgorithmNames.ToUpper(strName);
+                }
+                else
+                {
+                    // For desktop compat, exception is deferred until VerifySignature
+                    _algName = null;
+                }
             }
             catch (CryptographicException)
```

Comment:
It doesn't look like this Exception is achievable any more.  (Since the same try/catch is within the called method)